### PR TITLE
Upgrade Avalonia to 11.2.1

### PR DESCRIPTION
**Pull request by tomlm**
* I bumped version to 11.2.1
* I updated signature for changed methods
* implemented stubs for called methods which seem ok to ignore
* have not ImplementedException stubs for things I don't know if we need or not.
* Fixed some things that are obselete (RadiusX and RadiusY instead of Radius for example)

Everything seems to work except
* Menu
* ComboBox
* Dialog

They are blowing up inside KeyboardNavigationHandler, I don't know what those are/do, so I am pushing this branch up there. Maybe the solution is obvious to you?

### CODE REVIEW

File: src/Consolonia.Core/Drawing/ConsoloniaPlatformRenderInterfaceContext.cs [hunk 1]
```diff
@@ -20,6 +21,13 @@ public IRenderTarget CreateRenderTarget(IEnumerable<object> surfaces)
...
         }
 
+        public IDrawingContextLayerImpl CreateOffscreenRenderTarget(PixelSize pixelSize, double scaling)
+        {
+            throw new NotImplementedException();
+        }
+
         public bool IsLost => false;
+
+        public IReadOnlyDictionary<Type, object> PublicFeatures => new Dictionary<Type, object>();
```
Comment (jinek) [inline src/Consolonia.Core/Drawing/ConsoloniaPlatformRenderInterfaceContext.cs:31]: May be lets replace with just {get;} = new Dictionary<...>();  ? Just not to create an object on each call?

File: src/Consolonia.Core/Drawing/RenderTarget.cs [hunk 2]
```diff
@@ -69,7 +69,7 @@ void IDrawingContextLayerImpl.Blit(IDrawingContextImpl context)
 
         bool IDrawingContextLayerImpl.CanBlit => true;
 
-        public IDrawingContextImpl CreateDrawingContext()
+        public IDrawingContextImpl CreateDrawingContext(bool useScaledDrawing)
```
Comment (jinek) [inline src/Consolonia.Core/Drawing/RenderTarget.cs:72]: Shouldn't we throw NotImplemented when argument is true? Just to notify we do not support scaledDrawing.

File: src/Consolonia.Core/Infrastructure/ConsoleWindow.cs [hunk 3]
```diff
@@ -76,7 +76,7 @@ public IPopupImpl CreatePopup()
 
         public void SetTransparencyLevelHint(IReadOnlyList<WindowTransparencyLevel> transparencyLevels)
         {
-            throw new NotImplementedException("Consider this");
+            // I think we can ignore this.
```
Comment (jinek) [inline src/Consolonia.Core/Infrastructure/ConsoleWindow.cs:79]: Could you leave info when this happens? If well known

File: src/Consolonia.Core/Infrastructure/ConsoleWindow.cs [hunk 4]
```diff
@@ -134,7 +134,7 @@ public void Activate()
 
         public void SetTopmost(bool value)
         {
-            throw new NotImplementedException();
+            // todo
```
Comment (jinek) [inline src/Consolonia.Core/Infrastructure/ConsoleWindow.cs:137]: Leave please a comment when it was called

File: src/Consolonia.Themes.TurboVision/Templates/Controls/Helpers/ConsoloniaTextPresenter.cs [hunk 5]
```diff
@@ -71,10 +71,14 @@ public ConsoloniaTextPresenter()
         {
             // we need to disable blinking caret, our terminal caret blinks itself once shown
             var caretTickTimer = (DispatcherTimer)TickTimerField.GetValue(this);
-            caretTickTimer!.Interval =
-                TimeSpan.FromMilliseconds(int
-                    .MaxValue); //see DispatcherTimer.Interval, since we can not disable it, setting it to the longest interval possible
-            caretTickTimer!.Tick += (_, _) => throw new NotImplementedException("How to disable timer completely?");
+            // TODO: consider that with avalonia 11.1.5 this is null;
```
Comment (jinek) [inline src/Consolonia.Themes.TurboVision/Templates/Controls/Helpers/ConsoloniaTextPresenter.cs:74]: Probably we should investigate how to disable blinking in Avalonia 11.1.5 ?

File: src/Consolonia.Core/Drawing/ConsoloniaPlatformRenderInterfaceContext.cs (hunk 1 above)
Comment (tomlm) [reply src/Consolonia.Core/Drawing/ConsoloniaPlatformRenderInterfaceContext.cs:31]: This is a shorthand for that:
  private string _name;  // the name field
  public string Name => _name;     // the Name property with get only accessor

Comment (jinek) [reply src/Consolonia.Core/Drawing/ConsoloniaPlatformRenderInterfaceContext.cs:31]: I mean
```
public IReadOnlyDictionary<Type, object> PublicFeatures => new Dictionary<Type, object>();
```
is shorthand for
```
public IReadOnlyDictionary<Type, object> PublicFeatures
        {
            get
            {
                return new Dictionary<Type, object>();
            }
        }
```
New dictionary is initialized on each property call.

While 
```
public IReadOnlyDictionary<Type, object> PublicFeatures { get; } = new Dictionary<Type, object>();
```
is shorthand for
```
private readonly IReadOnlyDictionary<Type, object> _publicFeatures = new Dictionary<Type, object>();
        public IReadOnlyDictionary<Type, object> PublicFeatures => _publicFeatures;
```
where only one dictionary is initialized.

**Comment by jinek** [comment]
@tomlm Why did you decide to close this one?

File: src/Consolonia.Core/Drawing/ConsoloniaPlatformRenderInterfaceContext.cs (hunk 1 above)
Comment (tomlm) [reply src/Consolonia.Core/Drawing/ConsoloniaPlatformRenderInterfaceContext.cs:31]: got it, dang, that's subtle and dangerous. I'm going to have be a lot more careful using that pattern.

Comment (jinek) [reply src/Consolonia.Core/Drawing/ConsoloniaPlatformRenderInterfaceContext.cs:31]: Interesting thing that CodeRabbit did not notice this.

**Comment by tomlm** [comment]
I had a typo in the branch name and renaming the branch closed the PR!

**Pull request closed without merging**